### PR TITLE
Wmo 50065 generalise sensor reading on agv

### PR DESCRIPTION
**Pull request by moeed-logivations**
Make sensor base class
Add config options for sensors
Add yocto range finder as compatible sensor

### CODE REVIEW

**Comment by moeed-logivations** [comment]
@jplapp @arsentymchyshyn Could you please review this PR?

File: sensor_scripts/base_sensor.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    WARNING_FAR = 1
+    WARNING_FREE = 0
+    
+    def __init__(self):
+	super(BaseSensor, self).__init__()
+	self.warning_level = self.WARNING_EMERGENCY_STOP
+    
+    
+    def get_warning_level(self):
+        return self.warning_level
```
Comment (jplapp) [inline sensor_scripts/base_sensor.py]: reformat file

File: sensor_scripts/sick_sensor.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            warning_level = 3
+            return warning_level
+
+        def zfill(binary_res):
+            bins = bin(binary_res)
+            shortened = bins[2:]
+            return shortened.zfill(16)
+
+        sensor_reading = zfill(res)
+        print(sensor_reading)
```
Comment (jplapp) [inline sensor_scripts/sick_sensor.py]: remove print statement

File: sensor_scripts/yocto_rangefinder.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	self.start()
+    
+    def run(self):
+	while True:
+	    if not self.sensor.isOnline():
+		continue
+	    try:
+        	distance = self.sensor.get_currentValue()
+	    except Exception as e:
+		print("EXCEPTION [SENSOR]: ", e)
```
Comment (jplapp) [inline sensor_scripts/yocto_rangefinder.py]: use logging

File: sensor_scripts/base_sensor.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from threading import Thread
+
+
+# abstract base class
+class BaseSensor(Thread):
+    # warning macros
+    WARNING_EMERGENCY_STOP = 3
+    WARNING_CLOSE = 2
+    WARNING_FAR = 1
+    WARNING_FREE = 0
```
Comment (arsentymchyshyn) [inline sensor_scripts/base_sensor.py]: Can this global variables be moved out from class for global usage or just initialize them inside of init method?

File: sensor_scripts/yocto_rangefinder.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import sys
+import time
+from yoctopuce.yocto_rangefinder import *
```
Comment (arsentymchyshyn) [inline sensor_scripts/yocto_rangefinder.py]: too broad import we need to avoid this if possible

File: tcp_client.py [hunk 6]
```diff
@@ -1,110 +1,110 @@
...
+                cache = str_to_decode
+        return res, cache
+
+    def handle_write(self):
+        if self.data_to_send:
+            data_to_send = json.dumps(self.data_to_send)
+            data_to_send = str(len(data_to_send)) + '\n' + data_to_send
+            sent = self.send(data_to_send.encode('ascii'))
+            logger.info('%s', data_to_send)
+            self.data_to_send = data_to_send[sent:]  # can't keep half of a json object
```
Comment (arsentymchyshyn) [inline tcp_client.py]: has something been changed in this file?

File: sensor_scripts/yocto_rangefinder.py (hunk 5 above)
Comment (moeed-logivations) [reply sensor_scripts/yocto_rangefinder.py]: We're using too many things from this, so probably best to import *

File: tcp_client.py (hunk 6 above)
Comment (moeed-logivations) [reply tcp_client.py]: A couple of lines. Reformatting changed the indentation and thus the whole file.

File: sensor_scripts/base_sensor.py (hunk 4 above)
Comment (moeed-logivations) [reply sensor_scripts/base_sensor.py]: Done.

**Pull request merged**